### DISCUSSION TITLE
GitHub Actions Inputs Variables Not Working

### DISCUSSION THREAD

**Discussion by ArielSAdamsNASA** [community/community#26785] [Actions]
<p>I am having issues using <code>on.workflow_call.inputs</code>. I tried using the example from GitHub Docs, <a href="https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#onworkflow_callsecrets" class="inline-onebox" rel="noopener nofollow ugc">Workflow syntax for GitHub Actions - GitHub Docs</a>, and this also failed. The workflow is provided below:</p>
<pre><code class="lang-auto">name: "Test Inputs"

on:
  push:
  workflow_call:
    inputs:
      username:
        description: 'A username passed from the caller workflow'
        default: 'john-doe'
        required: false
        type: string
  
jobs:
  print-username:
    runs-on: ubuntu-latest

    steps:
      - name: Print the input name to STDOUT
        run: echo The username is ${{ inputs.username }}
</code></pre>
<p>This workflow only prints <code>The username is</code> when it should be <code>The username is john-doe</code>. I have also tried <code>${{ github.event.inputs.username }}</code> and <code>${{ INPUT_USERNAME}}</code>. Any help would be appreciated. Thank you!</p>

**Comment by Simran-B** [comment]
<p>Can you share the code of the other workflow that calls this workflow as well, please?</p>

**Comment by ArielSAdamsNASA** [comment]
<p>I just now tested the reusable workflow. It prints <code>The username is</code> when it should be <code>The username is john-doe</code> .</p>
<pre><code class="lang-auto">name: Reuse Workflow

on:
  push:

jobs:
  reuse:
    name: Reuse
    uses: ArielSAdamsNASA/cFS-JSF-Rules/.github/workflows/workflow.yml@Rule1
</code></pre>

**Comment by ArielSAdamsNASA** [comment, marked as answer]
<p>I resolved my issue. I was not using ‘’ for my inputs and did not realized that the original workflow will not use the default value itself. One must call the workflow first. Thank you for your help!</p>
